Using a file: link in a model so I can edit it locally causes an error.

**To Reproduce**
- goto Models
- for any model, click on the hamburger menu and select Edit
- under Links: add a link like file:///home/michael/foo.scad
- Save
- An Internal Server Error (500) message is displayed
- You have to select Edit and delete the link for the model to display again or you will keep getting the 500 error

**Expected behavior**
When I click on the link in Chrome in Manyfold xdg-open should just open the local file directly.

**Screenshots**

<img width="563" height="308" alt="Image" src="https://github.com/user-attachments/assets/45a00a7c-eb62-426c-addf-48b705adec27" />


**Desktop (please complete the following information):**
 - OS: Debian Linux 
 - Browser: Chrome
 - Version: Latest of both

**Additional context**
I want to edit files locally without having to download them in the browser and then re-upload the changes.
This is especially useful for OpenSCAD files since I often tweak them before printing a new version.

I have Chrome configured to open file:/// links instead of downloading them successfully.
But when I add a file:/// link to a model that model starts causing an error and I have to delete the link for it to work again.

Is there a better way to accomplish what I want to do?  Or if this is the way to do what I want, is the error fixable?

I'm not sure any other protocol names work besides http and https.

Ah, I didn't see how to pull logs, but here it is in case you just want to allow file: links:

app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] {"method":"GET","path":"/models/tl1dc71rtt4z","format":"html","controller":"ModelsController","action":"show","status":500,"allocations":38009,"duration":162.57,"view":0.0,"db":72.44,"@timestamp":"2026-06-28T14:26:10.338Z","@version":"1","message":"[500] GET /models/tl1dc71rtt4z (ModelsController#show)"}
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]   
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] ActionView::Template::Error (undefined method 'locale' for an instance of Hash)
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] Caused by: NoMethodError (undefined method 'locale' for an instance of Hash)
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] 
app-1  | 14:26:10 rails.1              | Information for: ActionView::Template::Error (undefined method 'locale' for an instance of Hash):
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     103:           <div><%= render "tag_list", tags: @model.tags.order(taggings_count: :desc, name: :asc), filter: @filter %></div>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     104:           <% unless @model.links.empty? %>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     105:             <div><%= Icon icon: "link-45deg", role: "presentation", label: Link.model_name.human(count: 100) %></div>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     106:             <%= LinkList links: @model.links, icons: false %>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     107:           <% end %>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     108:           <% if SiteSettings.social_enabled? %>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]     109:             <% if SiteSettings.federation_enabled? %>
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]   
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:246:in 'block in ApplicationHelper#translate_with_locale_wrapper'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:245:in 'ApplicationHelper#translate_with_locale_wrapper'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:23:in 'block (3 levels) in Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:19:in 'block (2 levels) in Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:17:in 'block in Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:16:in 'Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/components_helper.rb:7:in 'block (2 levels) in <module:ComponentsHelper>'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/views/models/show.html.erb:106
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:96:in 'block (3 levels) in ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:95:in 'block (2 levels) in ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:94:in 'block in ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:73:in 'ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/views/models/show.html.erb:61
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/controllers/application_controller.rb:141:in 'ApplicationController#switch_locale'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] 
app-1  | 14:26:10 rails.1              | Information for cause: NoMethodError (undefined method 'locale' for an instance of Hash):
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1]   
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:246:in 'block in ApplicationHelper#translate_with_locale_wrapper'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:245:in 'ApplicationHelper#translate_with_locale_wrapper'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:23:in 'block (3 levels) in Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:19:in 'block (2 levels) in Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:17:in 'block in Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/components/link_list.rb:16:in 'Components::LinkList#view_template'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/components_helper.rb:7:in 'block (2 levels) in <module:ComponentsHelper>'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/views/models/show.html.erb:106
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:96:in 'block (3 levels) in ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:95:in 'block (2 levels) in ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:94:in 'block in ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/helpers/application_helper.rb:73:in 'ApplicationHelper#card'
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/views/models/show.html.erb:61
app-1  | 14:26:10 rails.1              | [7cde2d3a-10fd-4546-a49a-552da9fb74c1] app/controllers/application_controller.rb:141:in 'ApplicationController#switch_locale'

--- a/app/models/link.rb
+++ b/app/models/link.rb
@@ -10,7 +10,7 @@ class Link < ApplicationRecord
   after_commit -> { linkable.try(:write_datapackage_later) }
 
   def host
-    URI.parse(url).host || url
+    URI.parse(url).host.presence || "link"
   rescue URI::InvalidURIError, URI::InvalidComponentError
     url
   end

